### ISSUE TITLE
Modal Dialog's "scroll lock" is incorrect

### ISSUE THREAD

**Issue by theKashey** [smooth-code/smooth-ui#25]
Modal dialog on mounting is setting `overflow:hidden` directly on body, and remove it on unmount.
Don't do that :)

Problems:
 1. What if you have two modals? It will "unlock" scroll after the first unmount.
 2. It will hide scroll bar on windows machines, casing content to "jump"
 3. It does not work for mobile safari at all 😺 

https://github.com/theKashey/react-scroll-locky will do the job

**Comment by neoziro** [comment]
Thanks, same I will take a look!

The problem with two modals is easy too solve, others are trickier!

**Comment by theKashey** [comment]
Just will the issues, and I'll help.

**Comment by theKashey** [comment]
- Add `<input type="text" />` into the modal
- Open it using Safari Mobile
- Select input
- ....
- Safari would use `position:absolute` instead of `position:fixed`, and your customer will lose the modal.

**Comment by neoziro** [comment]
Hello @theKashey, yeah I know the scroll lock is still incorrect 😣. Can you help on that point?

**Comment by theKashey** [comment]
Solved by creating another library :)
